test/test_persistent_cache.py

```diff
@@ -0,0 +1,117 @@
+import functools
+import os
+from subprocess import run, STDOUT, PIPE
+import sys
+import tempfile
+import unittest
+
+import torch_xla.runtime as xr
+
+
+# Wrapper to manage a temporary directory for the wrapped test
+def run_with_tmpdir(f):
+
+  @functools.wraps(f)
+  def run(*args, **kwargs):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      kwargs.setdefault('tmpdir', tmpdir)
+      f(*args, **kwargs)
+
+  return run
+
+
+# Basic command to generate a simple graph and perform metrics assertions
+METRICS_CMD_FMT = r'''
+import os
+import torch
+import torch_xla.core.xla_model as xm
+import torch_xla.debug.metrics as met
+import torch_xla.runtime as xr
+
+if 'TEST_WITH_SPMD' in os.environ:
+  xr.use_spmd()
+
+t = torch.randn(4, 4).to(xm.xla_device())
+s = t @ t
+xm.mark_step()
+
+for counter, value in %s:
+  actual = met.counter_value(counter)
+  assert actual == value, \
+    f'Unexpected value for counter {counter}: expected {value}, got {actual}'
+'''
+
+
+@unittest.skipUnless(xr.device_type() in {'TPU', 'GPU'},
+                     'Device type does not support persistent caching')
```

Comment:
why not cpu?

Reply:
I tested it, but CPU isn't supported (deserialization fails). JAX has a similar restriction: https://github.com/google/jax/blob/234be736c4cdd8da4197078278d35a6a1cde3767/tests/compilation_cache_test.py#L69C41-L69C41